### ISSUE TITLE
out-of-the-box generated modal re-renders when closing modal and clicking 'back' on browser

### ISSUE THREAD

**Issue by kuatroka** [phoenixframework/phoenix_live_view#3576]
### Environment

* Elixir version (elixir -v):

```
Erlang/OTP 27 [erts-15.1] [source] [64-bit] [smp:10:10] [ds:10:10:10] [async-threads:1] [jit]
Elixir 1.17.3 (compiled with Erlang/OTP 27)
```

* Phoenix version (mix deps):
`1.7.14`
* Phoenix LiveView version (mix deps):
`1.0.0`
* Operating system:
`Mac OS`
* Browsers you attempted to reproduce this bug on (the more the merrier): 
`Chrome, Safari, Brave`
* Does the problem persist after removing "assets/node_modules" and trying again? Yes/no:
**I don't have it.**
### Actual behavior

out-of-the-box generated modal re-renders when closing modal and clicking 'back' on browser. The modal was generated by the default generator `phx.gen.live` Everything works fine, but the modal re-rendering get's stuck in a loop when using the browser's **back** button. See gif.

![modal_repeat](https://github.com/user-attachments/assets/7cbd06b5-5a6f-44e5-8e8c-3c18c318f562)

P.S I'm very new to Phoenix. Here is my actual repo. it's not one page, but it comes with a SQLite DB saved in github too, so should work as is. When in the app, click on the **Analysis** in the main menu and then select one of **Countries, States, Cities, Territories List**. They all have the same issue and were all generated with the default `phx.gen.live`
[combobox repo](https://github.com/kuatroka/combobox)

### Expected behavior
When clicking on the **back** button it should take to the page where I was prior clicking on the page with the modal button and not to re launch the modal in a loop. Thanks

**Comment by kuatroka** [comment]
Thanks, but I'm not sure how to take it. I'm new to Elixir/Phoenix so bear with me pls.

I have used "phx.gen.live" and all these tables, functions and components were automatically created. I didn't specify any options other than the fields I need.
And now I've got a mini uk for the tables/list and when I click "edit" or "new" the modal pops up. I'm not sure then what it means "modal was removed"
Thanks

**Comment by LostKobrakai** [comment]
It means phoenix removed that code for newly generated phoenix projects. Core components become part of your codebase, so you can adjust them however you need to fix your problem. 

Also I think the expectation of the previous generates was indeed to have page navigation reopen the modal. It lived on it's own url that you can navigate from and to.

**Comment by kuatroka** [comment]
Thanks but weird. I've generated it a day ago and I'm using the latest liveview. I'll update to the latest Phoenix and see what happens

**Comment by SteffenDE** [comment]
@LostKobrakai is correct that the modal was removed. It is still there for new Phoenix 1.7.18 projects as the change will only be included in Phoenix 1.8.

That being said, the behavior as observed is expected. The modal has its own route in order to be reopened on client reconnects. You can change this behavior if you want, either by not having an extra route and opening the modal with the corresponding show/hide function purely clientside, by using an assign instead, or - and that’s probably the best solution I think - by using `replace: true` on navigation when closing the modal. Then, the back button should go to the previous page instead.

**Comment by kuatroka** [comment]
Thanks, I used your # 3 option, but it didn't work. It is still stuck in the loop. I'll check the others  later. Thanks again.
